src/test/java/io/fabric8/maven/docker/log/DefaultLogCallbackTest.java

```diff
@@ -0,0 +1,173 @@
+import io.fabric8.maven.docker.access.log.LogCallback.DoneException;
+import io.fabric8.maven.docker.util.Timestamp;
+
+public class DefaultLogCallbackTest {
+
+    private Path path;
+
+    private LogOutputSpec spec;
+
+    private LogCallback callback;
+
+    private Timestamp ts;
+
+    @Before
+    public void before() throws FileNotFoundException {
+        path = Paths.get("target", "one.log");
+        path.toFile().delete();
+        spec = new LogOutputSpec.Builder().prefix("callback-test")
+                .file(path.toString()).build();
+        callback = new DefaultLogCallback(spec);
+        ts = new Timestamp("2016-12-21T15:09:00.999666333Z");
+    }
+
+    @Test
+    public void shouldLogSequentially() throws IOException, DoneException {
+        callback.log(1, ts, "line 1");
+        callback.log(1, ts, "line 2");
+        callback.close();
+
+        List<String> lines = Files.readAllLines(path);
+        assertThat(lines, contains("callback-test> line 1", "callback-test> line 2"));
+    }
+
+    @Test
+    public void shouldLogError() throws IOException, DoneException {
+        callback.error("error 1");
+        callback.log(1, ts, "line 2");
+        callback.error("error 3");
+        callback.close();
+
+        List<String> lines = Files.readAllLines(path);
+        assertThat(lines, contains("error 1", "callback-test> line 2", "error 3"));
+    }
+
+    @Test
+    public void shouldLogToStdout() throws IOException, DoneException {
+        // we don't need the default stream for this test
+        callback.close();
+
+        path = Paths.get("target", "stdout.log");
+        FileOutputStream os = new FileOutputStream(path.toFile());
+        PrintStream ps = new PrintStream(os);
+        PrintStream stdout = System.out;
+        try {
+            System.setOut(ps);
+            spec = new LogOutputSpec.Builder().prefix("stdout")
+                    .build();
+            callback = new DefaultLogCallback(spec);
+            DefaultLogCallback callback2 = new DefaultLogCallback(spec);
+
+            callback.log(1, ts, "line 1");
+            callback2.log(1, ts, "line 2");
+            callback.log(1, ts, "line 3");
+            callback.close();
+            callback2.log(1, ts, "line 4");
+            callback2.close();
+
+            List<String> lines = Files.readAllLines(path);
+            assertThat(lines, contains("stdout> line 1", "stdout> line 2", "stdout> line 3", "stdout> line 4"));
+        } finally {
+            System.setOut(stdout);
+        }
+    }
+
+    @Test
+    public void shouldKeepStreamOpen() throws IOException, DoneException {
+        DefaultLogCallback callback2 = new DefaultLogCallback(spec);
+
+        callback.log(1, ts, "line 1");
+        callback2.log(1, ts, "line 2");
+        callback.log(1, ts, "line 3");
+        callback.close();
+        callback2.log(1, ts, "line 4");
+        callback2.close();
+
+        List<String> lines = Files.readAllLines(path);
```

Comment:
Unfortunately we can't use Java 8 and are still stuck to Java 7 for now (this might change in the future). So we need some other way to read all lines here.